paimon-format/src/main/java/org/apache/paimon/format/parquet/ParquetReaderFactory.java

```diff
@@ -393,7 +393,9 @@ public void recycle() {
         }
 
         public RecordIterator<InternalRow> convertAndGetIterator() {
-            result.set(columnarBatch.getNumRows());
+            result.reset(columnarBatch.getNumRows());
+            // Since we currently don't implement parquet's filter push down, row position is always
```

Comment:
Can we implement this correctly? I think maybe we will implement parquet's filter push down in future. Leaving a hole here can easily lead to bugs.